### PR TITLE
[regression-test](Variant) Add more cases related to schema changes

### PR DESCRIPTION
**Pull request by eldenmoon**
And fix bugs about schema change for variant:
fix bug schema change crash on doing schema change with tablet schema that contains extracted columns

## Proposed changes

Issue Number: close #xxx

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by eldenmoon** [comment]
run buildall

**Comment by eldenmoon** [comment]
run buildall

**Comment by eldenmoon** [comment]
run buildall

**Comment by eldenmoon** [comment]
run buildall

**Comment by eldenmoon** [comment]
run buildall

**Comment by eldenmoon** [comment]
run buildall

**Comment by eldenmoon** [comment]
run buildall

**Comment by eldenmoon** [comment]
run buildall

File: be/src/olap/schema_change.cpp [hunk 1]
```diff
@@ -467,10 +469,10 @@ Status LinkedSchemaChange::process(RowsetReaderSharedPtr rowset_reader, RowsetWr
 Status VSchemaChangeDirectly::_inner_process(RowsetReaderSharedPtr rowset_reader,
                                              RowsetWriter* rowset_writer,
                                              TabletSharedPtr new_tablet,
-                                             TabletSchemaSPtr base_tablet_schema) {
+                                             TabletSchemaSPtr base_tablet_schema,
+                                             TabletSchemaSPtr new_tablet_schema) {
     do {
-        auto new_block =
-                vectorized::Block::create_unique(new_tablet->tablet_schema()->create_block());
+        auto new_block = vectorized::Block::create_unique(new_tablet_schema->create_block());
```
Comment (xiaokang) [inline be/src/olap/schema_change.cpp:475]: what's the difference to new_tablet->tablet_schema() ?

**Review by xiaokang — COMMENTED** [review]
As discussed, remove the string,jsonb <-> variant sc temporarily and seperate mow partial update to another pr.

**Comment by eldenmoon** [comment]
run buildall

**Comment by eldenmoon** [comment]
run buildall

**Comment by eldenmoon** [comment]
run buildall

**Review by xiaokang — APPROVED** [review]
LGTM

**Review by airborne12 — APPROVED** [review]
LGTM

**Pull request merged**